manifests/ipa/install.pp

```diff
@@ -0,0 +1,121 @@
+# Run ipa-client-install on puppet clients
+#
+# @param ensure
+#   'present' to install, 'absent' to uninstall
+#
+# @param enroll
+#   Be smart about when to run `ipa-client-install`
+#
+# @param ip_address
+#   IP address of host being connected
```

Comment:
Why is this undef by default instead of using a fact?  More explanation is needed.

Reply:
It is not required to pass the `--ip-address` option to `ipa-client-install`, so I didn't want to set it to anything by default. Same goes for `--hostname` below. I will add more documentation describing this.